### PR TITLE
Added solution to Leetcode 36(medium) , 37(hard) problems

### LINKED ISSUE #806: [Feature Request]: Add Leetcode solution for problem 36 and 37 : Valid Sudoku and Sudoku Solver solution
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

The solution to the problem is missing 

### Use Case

Solutions to all the problems if added, will enhance the projects use case

### Benefits

The solution to the problem is missing . Solutions to all the problems if added, will enhance the projects use case.

### Add ScreenShots

--

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by debangi29**
## Related Issue

Fix #806 
[Cite any related issue(s) this pull request addresses. If none, simply state "None”]

## Description

[Please include a brief description of the changes or features added]

## Type of PR

- [x] Feature enhancement

## Screenshots / Videos (if applicable)

image

image

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

**Comment by debangi29** [comment]
Hello @Ajay-Dhangar 
I wanted to ask that, pull requests solving one leetcode problem has level 2
This PR solve one easy and one hard problem together
Doesn't it deserve a level 3?
Please check it once...
Thanks a lot :)

File: dsa-solutions/lc-solutions/0000-0099/0036-Valid-Sudoku.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+## Problem Description
+
+| Problem Statement | Solution Link | LeetCode Profile |
+| :---------------- | :------------ | :--------------- |
+| [Valid Sudoku](https://leetcode.com/problems/valid-sudoku/description/) | [Valid Sudoku Solution on LeetCode](https://leetcode.com/problems/valid-sudoku/solutions/) |  [Debangi Ghosh](https://leetcode.com/u/debangi_29/) |
+
+## Problem Description
+
+Determine if a 9 x 9 Sudoku board is valid. Only the filled cells need to be validated according to the following rules:
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0036-Valid-Sudoku.md:20]: replace `9 x 9` to `$9 \times 9$`

File: dsa-solutions/lc-solutions/0000-0099/0036-Valid-Sudoku.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+- Each row must contain the digits 1-9 without repetition.
+- Each column must contain the digits 1-9 without repetition.
+- Each of the nine 3 x 3 sub-boxes of the grid must contain the digits 1-9 without repetition.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0036-Valid-Sudoku.md:24]: replace `3 x 3` to `$3 \times 3$`

File: dsa-solutions/lc-solutions/0000-0099/0036-Valid-Sudoku.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+Note:
+
+A Sudoku board (partially filled) could be valid but is not necessarily solvable.
+Only the filled cells need to be validated according to the mentioned rules.
+
+### Examples
+
+#### Example 1:
+
+**Input**: board =
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0036-Valid-Sudoku.md:34]: ### Bad practice

```md
**Input**: board = 
<br>
[["5","3",".",".","7",".",".",".","."]<br>
,["6",".",".","1","9","5",".",".","."]<br>
,[".","9","8",".",".",".",".","6","."]<br>
,["8",".",".",".","6",".",".",".","3"]<br>
,["4",".",".","8",".","3",".",".","1"]<br>
,["7",".",".",".","2",".",".",".","6"]<br>
,[".","6",".",".",".",".","2","8","."]<br>
,[".",".",".","4","1","9",".",".","5"]<br>
,[".",".",".",".","8",".",".","7","9"]]<br>
```

### Good Practice

````md
```
Input: board = 
[["5","3",".",".","7",".",".",".","."]
,["6",".",".","1","9","5",".",".","."]
,[".","9","8",".",".",".",".","6","."]
,["8",".",".",".","6",".",".",".","3"]
,["4",".",".","8",".","3",".",".","1"]
,["7",".",".",".","2",".",".",".","6"]
,[".","6",".",".",".",".","2","8","."]
,[".",".",".","4","1","9",".",".","5"]
,[".",".",".",".","8",".",".","7","9"]]
```
````

File: dsa-solutions/lc-solutions/0000-0099/0036-Valid-Sudoku.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+,["4",".",".","8",".","3",".",".","1"]<br>
+,["7",".",".",".","2",".",".",".","6"]<br>
+,[".","6",".",".",".",".","2","8","."]<br>
+,[".",".",".","4","1","9",".",".","5"]<br>
+,[".",".",".",".","8",".",".","7","9"]]<br>
+
+**Output**: true
+#### Example 2:
+
+- **Input:** board =
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0036-Valid-Sudoku.md:49]: and also, change it

File: dsa-solutions/lc-solutions/0000-0099/0037-Sudoku-Solver.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+The '.' character indicates empty cells.
+
+ 
+
+### Examples
+
+#### Example 1:
+
+**Input**: board = 
+<br>
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0037-Sudoku-Solver.md:38]: also, change for good practice

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
now update your code based on suggestions

**Comment by debangi29** [comment]
I have done the changes

Thanks :)

**Comment by debangi29** [comment]
Should I do any more changes?
@Ajay-Dhangar

File: dsa-solutions/lc-solutions/0000-0099/0036-Valid-Sudoku.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+,[".","9","8",".",".",".",".","6","."]
+,["8",".",".",".","6",".",".",".","3"]
+,["4",".",".","8",".","3",".",".","1"]
+,["7",".",".",".","2",".",".",".","6"]
+,[".","6",".",".",".",".","2","8","."]
+,[".",".",".","4","1","9",".",".","5"]
+,[".",".",".",".","8",".",".","7","9"]]
+```
+**Output:**  false
+<br>
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0036-Valid-Sudoku.md:62]: remove `<br>` tag

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
everything is fine but you remove `<br>` tag and solve error

File: dsa-solutions/lc-solutions/0000-0099/0036-Valid-Sudoku.md (hunk 6 above)
Comment (Ajay-Dhangar) [reply dsa-solutions/lc-solutions/0000-0099/0036-Valid-Sudoku.md:62]: @debangi29 complete your task

**Comment by debangi29** [comment]
I have done it @Ajay-Dhangar

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**